libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Analyzer.cs

```diff
@@ -141,17 +145,45 @@ public override List<string> VisitNormalTemplateString([NotNull] LGFileParser.No
         private List<string> AnalyzeExpression(string exp)
         {
             exp = exp.TrimStart('{').TrimEnd('}');
-            var parseTree = ExpressionEngine.Parse(exp);
-            return AnalyzeParserTree(parseTree);
-        }
-
-        private List<string> AnalyzeParserTree(IParseTree parserTree)
-        {
-            var result = new List<string>();
-            
-            var visitor = new ExpressionAnalyzerVisitor(Context);
-
-            return visitor.Analyzer(parserTree);
+            var parse = _expressionParser.Parse(exp);
+            var references = new HashSet<string>();
+            var path = Microsoft.Bot.Builder.Expressions.Extensions.ReferenceWalk(parse, references,
```

Comment:
I' d add this to a using

Reply:
What are you suggesting for the using?  The parser has no resources when you create it--only when you parse and they are scoped to the parse method.  If you are talking expressions, they are not heavyweight.